app/src/main/java/io/neurolab/network/OSCReceiverFragment.java

```diff
@@ -0,0 +1,86 @@
+package io.neurolab.network;
+
+import android.content.Context;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentActivity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.FrameLayout;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import java.net.SocketException;
+import java.util.ArrayList;
+
+import io.neurolab.R;
+
+public class OSCReceiverFragment extends FragmentActivity {
+    private LinearLayout listenerTextList;
+    private ListView messageOutputTextList;
+    private FrameLayout frameLayout;
+    private TextView statusLabel;
+    private EditText ipField;
+    private EditText portField;
+    private Button forwardButton;
+
+    private OSCReceiver oscReceiver;
+    private ArrayList<String> listenerList;
+
+    public OSCReceiverFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        Bundle bundle = getIntent().getExtras();
+        super.onCreate(savedInstanceState);
+        listenerTextList = findViewById(R.id.listenerTextList);
+        messageOutputTextList = findViewById(R.id.messageOutputTextList);
+        frameLayout = findViewById(R.id.frame);
+        statusLabel = findViewById(R.id.statuslabel);
+        ipField = findViewById(R.id.ipField);
+        portField = findViewById(R.id.portField);
+        forwardButton = findViewById(R.id.forwardButton);
+
+        getItemList();
+
+        forwardButton.setOnClickListener(v -> {
+            try {
+                OSCReceiver.run( bundle.getString("filename"));
+                oscReceiver = new OSCReceiver(Integer.parseInt(String.valueOf(portField.getText())), listenerList);
+            } catch (SocketException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    private void getItemList() {
+        listenerList = new ArrayList<>();
+        for(int i = 0;i<listenerTextList.getChildCount();i++){
```

Comment:
Please reformat the code before committing

Reply:
Done